### DISCUSSION TITLE
500: INTERNAL_SERVER_ERROR

### DISCUSSION THREAD

**Discussion by RayyanAzka** [vercel/community#2195] [Help] [answered]
### Summary

I got this error and I don't know how to solve this. 
![Screenshot 2023-04-26 082107](https://user-images.githubusercontent.com/70903194/234504293-5c00e585-2aae-46fa-af33-d5be7732bc0a.png)
![Screenshot 2023-04-26 143708](https://user-images.githubusercontent.com/70903194/234504307-9c21c8de-0160-4fed-8e85-06b8fa04651d.png)
![Screenshot 2023-04-26 082050](https://user-images.githubusercontent.com/70903194/234504245-23c5c8bc-2b26-41c5-a6f5-5d995b54c907.png)
Any solutions?

**Comment by amyegan** [comment]
Hi @RayyanAzka! The [FUNCTION_INVOCATION_FAILED](https://vercel.com/docs/errors/application-error-codes#function_invocation_failed) error is often caused by an unhandled exception. One strategy for debugging is to wrap your function code in a `try` and `except` block and log the exception. That should give you more details in the function logs

**Reply by RayyanAzka** [reply]
so should I write try-except inside wsgi.py?

**Reply by amyegan** [reply]
I'm not familiar enough with writing Python apps to recommend anything specific. 😓 But if we were working with a JavaScript serverless function, it would look something like this:
```javascript
export default function handler(request, response) {
  try {
    response.status(200).json({message: "Hello, world!"});
  } catch(error) {
    console.log(error);
  }
}
```

It's also a good idea to review the documentation on [using the Python Runtime with Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python#) to be sure you have everything set up for Vercel to run your code properly. It's a bit different from a traditional server setup.

**Reply by RayyanAzka** [reply]
I think I just found the cause, but I don't understand the solution to solve this cause

Error: Cannot read properties of undefined (reading 'includes')
Error: An unexpected error occurred!
DeploymentError: Too many requests - try again in 6 hours (more than 5000, code: "api-upload-free").
    at uploadList.<computed>.async_retry_1.default.retries (/usr/lib/node_modules/vercel/dist/index.js:18[52](https://gitlab.com/RayyanAzka/propensi-d02-dummy/-/jobs/4185487489#L52)83:27)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Error: An unexpected error occurred!
DeploymentError: Too many requests - try again in 6 hours (more than 5000, code: "api-upload-free").
    at uploadList.<computed>.async_retry_1.default.retries (/usr/lib/node_modules/vercel/dist/index.js:185283:27)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Reply by amyegan** [reply, marked as answer]
That looks like you're running into [Rate Limits](https://vercel.com/docs/concepts/limits/overview#rate-limits). In this case, it seems to be the [5000 uploads per day](https://vercel.com/docs/concepts/limits/overview#:~:text=Uploads%20per%20day%20(Free)) limit. 

This limit can be reached if you have a large number of files in a single deployment or across multiple deployments in a single day. If you have a lot of files in the project that do not need to be included in the deployment, you can [exclude files with .vercelignore](https://vercel.com/docs/concepts/deployments/vercel-ignore#)

**Reply by sourabhverma05** [reply]
@amyegan please solve this in next js
